### PR TITLE
Remove angular.uppercase/lowercase

### PR DESCRIPTION
**Pull request by Narretz**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
fix/refactor

**What is the current behavior? (You can also link to an open issue here)**
we expose lowercase/uppercase on the public angular object

**Does this PR introduce a breaking change?**
yes

**Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines: https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#commit-message-format
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Other information**:

### CODE REVIEW

File: src/ngMock/angular-mocks.js [hunk 1]
```diff
@@ -2072,13 +2072,13 @@ function MockXhr() {
     var header = this.$$respHeaders[name];
     if (header) return header;
 
-    name = angular.lowercase(name);
+    name = angular.$$lowercase(name);
```
Comment (mgol) [inline src/ngMock/angular-mocks.js]: Why keep the service, even as a private one? `String.prototype` methods should be enough.

**Review by mgol — DISMISSED** [review]

File: src/ngMock/angular-mocks.js (hunk 1 above)
Comment (Narretz) [inline src/ngMock/angular-mocks.js]: I just noticed that angular.lowercase / uppercase is overwritten with a manual version that handles Turkish `i`: https://github.com/angular/angular.js/blob/19bc52127f8393ffccac563b21267adb75c19275/src/Angular.js#L210-L232

So in core, we use a "safe" version. But we don't want anyone else to use it? I guess we don't want anyone to use the special version when usually the prototype functions are enough.

Comment (mgol) [inline src/ngMock/angular-mocks.js]: jQuery doesn't have any logic like that in its lowercasing and we haven't heard any complaints related to Turkish keyboards. Given jQuery's popularity I'd assume this code is no longer needed, especially that ES5 mandates no such differences so I doubt any relatively recent browser would still have whatever issue existed (if it ever has).

**Comment by petebacondarwin** [comment]
LGTM - keeping the private methods for the time being.

**Comment by Narretz** [comment]
K. FWIW, the "problem" still exists in modern browsers: http://plnkr.co/edit/cQHWgcqTLfZZ2LZXQQqG?p=preview 
I can't really test toLocaleLowerCase correctly because the browser must be turkish ...

**Comment by mgol** [comment]
@Narretz I don't see a problem in your fiddle. `toLowerCase` should always work in the same way so the Turkish lower "i" won't be the same as lowercased regular uppercase "I"; that seems expected.

**Comment by Narretz** [comment]
Hm. So if we use regular lowercase, then theoretically there could be a case where we lowercase some developer / user provided value, which is then sent to a server and arrives "incorrectly" lower cased when compared to the expected lower case string on the server. I believe we do this to http headers, for example.

**Comment by mgol** [comment]
The `manualLowercase` function won't help in such a case either as it only lowercases Latin letters (from `A` to `Z`). This is, BTW, different from the native `String#toLowerCase`:
http://plnkr.co/edit/dZr9bnG8nIboMjwpgMpW?p=preview

So if there is any browser that violates ES5 and with Turkish locale has `'i' !== 'I'.toLowerCase()` then we'll use a lowercasing function that doesn't convert lots of letters that we convert otherwise. Wouldn't it cause issues?

**Pull request merged**